File: app/src/main/java/mishka/mingeo/data/model/DateConverter.java
package mishka.mingeo.data.model;

import android.arch.persistence.room.TypeConverter;

import org.joda.time.Instant;

public class DateConverter {
    @TypeConverter
    public static long instantToMillis(Instant instant){
        return instant.getMillis();
    }
    @TypeConverter
    public static Instant millisToInstant(long millis){
        return new Instant(millis);
    }
}

File: app/src/main/java/mishka/mingeo/data/model/DateConverter.kt
package mishka.mingeo.data.model

import android.arch.persistence.room.TypeConverter

import org.joda.time.Instant

class DateConverter {
    @TypeConverter
    fun instantToMillis(instant: Instant?): Long = instant?.millis ?: -1

    @TypeConverter
    fun millisToInstant(millis: Long): Instant? = if (millis == -1L) null else Instant(millis)

}
